cantabular/static_dataset.go

```diff
@@ -153,3 +153,91 @@ func (c *Client) StaticDatasetQueryStreamCSV(ctx context.Context, req StaticData
 	// Stream is responsible for closing the response body
 	return rowCount, stream.Stream(ctx, res.Body, transform, consume)
 }
+
+// Checks the number of observations returned from a cantabular query
+func (c *Client) CheckQueryCount(ctx context.Context, req StaticDatasetQueryRequest) (int, error) {
+	data := QueryData{
+		Dataset:   req.Dataset,
+		Variables: req.Variables,
+		Filters:   req.Filters,
+	}
+
+	logData := log.Data{
+		"url":     fmt.Sprintf("%s/graphql", c.extApiHost),
+		"request": req,
+	}
+
+	var q struct {
+		Data   StaticDatasetQuery `json:"data"`
+		Errors []gql.Error        `json:"errors"`
+	}
+
+	if err := c.queryUnmarshal(ctx, QueryStaticDataset, data, &q); err != nil {
+		return 0, dperrors.New(
+			fmt.Errorf("failed to make GraphQL query: %w", err),
+			http.StatusInternalServerError,
+			logData,
+		)
+	}
+
+	if len(q.Data.Dataset.Table.Error) > 0 || q.Data.Dataset.Table.Values == nil {
+		var err string
+		if len(q.Errors) > 0 {
+			err = q.Errors[0].Message
+		}
+
+		if len(q.Data.Dataset.Table.Error) > 0 {
+			err = q.Data.Dataset.Table.Error
+		}
+		return 0, dperrors.New(
+			errors.New(c.parseTableError(err)),
+			http.StatusBadRequest,
+			logData,
+		)
+	}
+
+	if len(q.Errors) > 0 {
+		return 0, dperrors.New(
+			errors.New("error(s) returned by graphQL query"),
+			q.Errors[0].StatusCode(),
+			logData,
+		)
+	}
+
+	return len(q.Data.Dataset.Table.Values), nil
+}
+
+// StaticDatasetQueryStreamJson performs a StaticDatasetQuery call
+// and then starts 2 go-routines to transform the response body into a Json stream and
+// consume the transformed output with the provided Consumer concurrently.
+// Returns a json formatted response
+// Use this method if large query responses are expected.
+func (c *Client) StaticDatasetQueryStreamJson(ctx context.Context, req StaticDatasetQueryRequest, consume Consumer) (GetObservationsResponse, error) {
```

Comment:
Is there a unit test for this?

Reply:
This is fully tested as part of component tests in dp-population-types-api that consumes it.